frontend/src/metabase/components/EntityItem.jsx

```diff
@@ -22,6 +22,139 @@ const EntityItemWrapper = Flex.extend`
   }
 `;
 
+function getPinnedBackground(model) {
+  return model === "dashboard"
+    ? color("accent4")
+    : lighten(color("accent4"), 0.28);
+}
+
+function getPinnedForeground(model) {
+  return model === "dashboard" ? color("white") : color("accent4");
+}
+
+function getBackground(model) {
+  return model === "dashboard" ? color("brand") : color("brand-light");
+}
+
+function getForeground(model) {
+  return model === "dashboard" ? color("white") : color("brand");
+}
+
+function EntityItemIcon({
+  item,
+  variant,
+  iconName,
+  pinned,
+  selectable,
+  selected,
+  onToggleSelected,
+  ...props
+}) {
+  const iconSize = variant === "small" ? 12 : 18;
+  const handleClick = e => {
+    e.preventDefault();
+    onToggleSelected();
+  };
+
+  return (
+    <IconWrapper
+      p={"12px 13px"}
+      mr={2}
```

Comment:
Let's remove `mr` and pass it in when needed since layout shouldn't be a concern of this component